src/Notes/BeforeCampaignMigration.php

```diff
@@ -0,0 +1,79 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Notes;
+
+use Automattic\WooCommerce\Admin\Notes\Note as NoteEntry;
+use Automattic\WooCommerce\GoogleListingsAndAds\Ads\AdsAwareTrait;
+use Automattic\WooCommerce\GoogleListingsAndAds\HelperTraits\Utilities;
+use Automattic\WooCommerce\GoogleListingsAndAds\Ads\AdsAwareInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\PluginHelper;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class BeforeCampaignMigration
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Notes
+ */
+class BeforeCampaignMigration extends AbstractNote implements AdsAwareInterface {
+
+	use AdsAwareTrait;
+	use PluginHelper;
+	use Utilities;
+
+	/**
+	 * Get the note's unique name.
+	 *
+	 * @return string
+	 */
+	public function get_name(): string {
+		return 'gla-before-campaign-migration';
+	}
+
+	/**
+	 * Get the note entry.
+	 */
+	public function get_entry(): NoteEntry {
+		$note = new NoteEntry();
+
+		$note->set_title( __( 'Your Google Listings & Ads campaigns will soon be automatically upgraded', 'google-listings-and-ads' ) );
+		$note->set_content(
+			__(
+				'From July through September, Google will be upgrading your existing campaigns from Smart Shopping to Performance Max, giving you the same benefits, plus expanded reach. There will be no impact to your spend or campaign settings due to this upgrade.',
+				'google-listings-and-ads'
+			)
+		);
+		$note->set_content_data( (object) [] );
```

Comment:
❓ Just curious. What is this for?

Reply:
`set_content_data` is used to add extra information to the Inbox Notification, in this case, we do not need to add extra information so I was passing an empty object as per other Notes in our codebase, however, looking at the default values of Notes::class this field is not needed because the default value is an empty object.

See more here: 

> In addition to content, notes also support structured content.  You can use this property to re-localize notes on the fly, but that is just one use.  You can store other data here too.  This is backed by a longtext column in the database.

Reference: https://developer.woocommerce.com/2020/10/16/using-the-admin-notes-inbox-in-woocommerce/